Source/WebCore/css/parser/CSSPropertyParserConsumer+Ident.h

```diff
@@ -106,6 +106,12 @@ std::optional<typename Map::ValueType> peekIdentUsingMapping(CSSParserTokenRange
 
 String consumeCustomIdentRaw(CSSParserTokenRange&, bool shouldLowercase = false);
 RefPtr<CSSPrimitiveValue> consumeCustomIdent(CSSParserTokenRange&, bool shouldLowercase = false);
+inline RefPtr<CSSPrimitiveValue> consumeCustomIdentExcluding(CSSParserTokenRange& range, std::initializer_list<CSSValueID> excluding, bool shouldLowercase = false)
+{
+    if (std::ranges::any_of(excluding, [peekedValueID = range.peek().id()](const auto& valueID) { return valueID == peekedValueID; }))
+        return nullptr;
+    return consumeCustomIdent(range, shouldLowercase);
+}
```

Comment:
I slightly prefer to have both a declaration and a separate definition, so the implementation is not interspersed with the interface. I haven’t looked over the whole file to see if that would enhance things here, but I’d be tempted to do it that way.

Reply:
The file is a pretty bad offender, but its not reason not to fix it up.